src/data/matrix/basic.lean

```diff
@@ -1032,6 +1032,31 @@ begin
   ext i j, rcases i; rcases j; simp [from_blocks],
 end
 
+/-- Let `p` pick out certain rows and `q` pick out certain columns of a matrix `M`. Then
+  `to_block M p q` is the corresponding block matrix. -/
+def to_block (M : matrix m n α) (p : m → Prop) [decidable_pred p]
+  (q : n → Prop) [decidable_pred q] : matrix {a // p a} {a // q a} α := λ i j, M ↑i ↑j
```

Comment:
I think this might be `:= M.minor coe coe`, although I don't know if that makes things easier or harder for you.

Reply:
Wow...

Reply:
So I can state
```
lemma to_block_matrix_det (M : matrix m m R) (p : m → Prop) [decidable_pred p] :
  M.det = (matrix.from_blocks (to_block M p p) (to_block M p (λ j, ¬p j))
    (to_block M (λ j, ¬p j) p) (to_block M (λ j, ¬p j) (λ j, ¬p j))).det
```
like this:
```
lemma to_block_matrix_det (M : matrix m m R) (p : m → Prop) [decidable_pred p] :
  M.det =
    (matrix.from_blocks (minor M coe coe) (minor M coe coe) (minor M coe coe) (minor M coe coe) :
     matrix ({a // p a} ⊕ {a // ¬p a}) ({a // p a} ⊕ {a // ¬p a}) R).det
```
Is that worth it?

Reply:
To be clear, I'm just suggesting you implement these definitions in terms of the existing one, not that you remove them.

Reply:
But `M.minor coe coe` is longer than `λ i j, M ↑i ↑j` and harder to parse (for humans and, probably, lean). Is there some benefit I'm missing? `M.minor` doesn't have much of an API, in fact 
```
@[simp] lemma minor_apply (A : matrix m n α) (row : l → m) (col : o → n) (i : l) (j : o) :
  minor A row col i j = A (row i) (col j) := rfl
```
should probably be added...

Reply:
Even if it has no API today, if you build your API on top of it, then it makes it easier to apply whatever API `minor` inherits later.

Reply:
I made this change